pkg/controller/operators/olm/operator.go

```diff
@@ -241,6 +279,212 @@ func (a *Operator) syncRBAC(obj interface{}) (syncError error) {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	// TODO(jpeeler): this needs to use user impersonation with op.Spec.ServiceAccount
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+
+	nsCount := len(namespaceList.Items)
+
+	if len(op.Status.Namespaces) == nsCount {
+		nsMap := map[string]struct{}{}
+		for _, v := range namespaceList.Items {
+			nsMap[v.Name] = struct{}{}
+		}
+		equalityCheck := true
+		for _, v := range op.Status.Namespaces {
+			if _, ok := nsMap[v.Name]; !ok {
+				equalityCheck = false
+				break
+			}
+		}
+		if equalityCheck {
+			// status is current with correct namespaces, so no further updates required
+			return nil
+		}
+	}
+	op.Status.Namespaces = namespaceList.Items
+	op.Status.LastUpdated = timeNow()
+
+	// make string list made up of comma separated namespaces
+	var nsList strings.Builder
+	for i := 0; i < nsCount-1; i++ {
+		nsList.WriteString(namespaceList.Items[i].Name + ",")
+	}
+	nsList.WriteString(namespaceList.Items[nsCount-1].Name)
+
+	currentNamespace := op.GetNamespace()
+	csvsInNamespace := a.csvsInNamespace(currentNamespace)
+	for csvName, csv := range csvsInNamespace {
+		strategy, err := a.resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+		if err != nil {
+			return fmt.Errorf("error unmarshaling strategy from ClusterServiceVersion '%s' with error: %s", csvName, err)
+		}
+
+		strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+		if !ok {
+			return fmt.Errorf("could not assert strategy implementation as deployment for CSV %s", csvName)
+		}
+
+		managerPolicyRules := []rbacv1.PolicyRule{}
+		apiEditPolicyRules := []rbacv1.PolicyRule{}
+		apiViewPolicyRules := []rbacv1.PolicyRule{}
+		for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+			resourceNames := []string{}
+			for _, resource := range owned.Resources {
+				resourceNames = append(resourceNames, resource.Name)
+			}
+			managerPolicyRules = append(managerPolicyRules, rbacv1.PolicyRule{Verbs: []string{"*"}, APIGroups: []string{owned.Name}, Resources: resourceNames})
+			apiEditPolicyRules = append(apiEditPolicyRules, rbacv1.PolicyRule{Verbs: []string{"create", "update", "patch", "delete"}, APIGroups: []string{owned.Name}, Resources: []string{owned.Kind}})
+			apiViewPolicyRules = append(apiViewPolicyRules, rbacv1.PolicyRule{Verbs: []string{"get", "list", "watch"}, APIGroups: []string{owned.Name}, Resources: []string{owned.Kind}})
+		}
+		clusterRole := &rbacv1.ClusterRole{
+			Rules: managerPolicyRules,
+		}
+		ownerutil.AddNonBlockingOwner(clusterRole, csv)
+		clusterRole.SetGenerateName(fmt.Sprintf("owned-crd-manager-%s-", csv.Spec.DisplayName))
+		createdClusterRole, err := a.OpClient.KubernetesInterface().RbacV1().ClusterRoles().Create(clusterRole)
+		if err != nil {
+			return err
+		}
+		for _, sa := range strategyDetailsDeployment.Permissions {
```

Comment:
These generated roles aren't for the serviceaccount for the operator to use, they're for end users and administrators.

It's so that a cluster admin can bind `jpeeler` to `etcd-edit` in namespace `foo` to give access to operator-provided services.

(we will need a separate thing to ensure that the serviceaccount for the operator has the correct permissions in target namespaces)